### PR TITLE
DOC: read_excel doc - fixed formatting and added examples

### LINKED ISSUE #18735: read_excel kwarg for `comment`?
This is a feature request to add a `comment` kwarg to `pandas.read_excel` just like `pandas.read_csv` can accept a `comment` kwarg.

Sorry if this request should be placed somewhere else.

### PR DESCRIPTION
**Pull request by JanLauGe**
Fixes a formatting bug in the `read_excel` docs that caused a line break and bold print in list of `_NA_VALUES`. Adds examples in the `read_excel` docstring.

- [x] closes #18735 
- [x] tests added & passed
- [x] passes `git diff master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by JanLauGe** [comment]
First `pandas` PR, please let me know about anything that should be done differently.
Thank you!

**Comment by chris-b1** [comment]
Thanks @JanLauGe! to close #18735 I'd like to see the following

1.  add `comment` in code to the explicit named keyword arguments to the `read_excel` function, and subsequent internal calls

2.  add `comment` to the docstring for the same

3.  add a few tests for `comment` functionality

**Comment by JanLauGe** [comment]
Thanks @chris-b1 ! To make sure I understand: You'd like `comment` to be a named argument instead of a keyword argument?

File: pandas/tests/io/test_excel.py [hunk 1]
```diff
@@ -1862,6 +1862,62 @@ def test_invalid_columns(self):
             with pytest.raises(KeyError):
                 write_frame.to_excel(path, 'test1', columns=['C', 'D'])
 
+    def test_comment_arg(self):
+        # Test the comment argument functionality to read_excel
```
Comment (jreback) [inline pandas/tests/io/test_excel.py]: add issue number here as a comment

File: pandas/tests/io/test_excel.py [hunk 2]
```diff
@@ -1862,6 +1862,62 @@ def test_invalid_columns(self):
...
+
+            # Create file to read in
+            write_frame = DataFrame({'A': ['one', '#one', 'one'],
+                                     'B': ['two', 'two', '#two']})
+            write_frame.to_excel(path, 'test_c')
+
+            # Read file without comment arg
+            read_frame = read_excel(path, 'test_c')
+            read_frame_commented = read_excel(path, 'test_c', comment='#')
+            tm.assert_class_equal(read_frame, read_frame_commented)
```
Comment (jreback) [inline pandas/tests/io/test_excel.py]: be consistent with other tests on nameing, e.g. 
``write_frame -> df``
``result = read_excel``


use ``assert_frame_equal``

File: pandas/tests/io/test_excel.py [hunk 3]
```diff
@@ -1862,6 +1862,62 @@ def test_invalid_columns(self):
...
+
+    def test_comment_default(self):
+        # Test the comment argument default to read_excel
+        with ensure_clean(self.ext) as path:
+
+            # Create file to read in
+            write_frame = DataFrame({'A': ['one', '#one', 'one'],
+                                     'B': ['two', 'two', '#two']})
+            write_frame.to_excel(path, 'test_c')
```
Comment (jreback) [inline pandas/tests/io/test_excel.py]: same on naming

File: pandas/tests/io/test_excel.py [hunk 4]
```diff
@@ -1862,6 +1862,62 @@ def test_invalid_columns(self):
...
+    def test_comment_emptyline(self):
+        # Test that read_excel ignores commented lines at the end of file
+        with ensure_clean(self.ext) as path:
+
+            write_frame = DataFrame({'a': ['1', '#2'], 'b': ['2', '3']})
+            write_frame.to_excel(path, index=False)
+
+            # Test that all-comment lines at EoF are ignored
+            read_frame_short = read_excel(path, comment='#')
+            assert (read_frame_short.shape == write_frame.iloc[0:1, :].shape)
```
Comment (jreback) [inline pandas/tests/io/test_excel.py]: comapre using ``assert_frame_equal``

**Review by jreback — CHANGES_REQUESTED** [review]
nice examples. pls add a whatsnew note in 0.22.0, you can put this in other api changes section, just say comment arg is exposed as a named parameter in read_excel.

**Review by chris-b1 — APPROVED** [review]
lgtm with @jreback's comments addressed

**Comment by JanLauGe** [comment]
Thanks for the comments @jreback, that should be all done now.
Please let me know if there is anything else outstanding.

**Comment by alysivji** [comment]
I spent a bunch of time lining up the arguments in read_excel to match read_csv

Would it be possible to keep following this pattern? Looks like `comment` can go between `thousands` and `skip_footer`

**Comment by JanLauGe** [comment]
Good point @alysivji, correct now?

**Comment by alysivji** [comment]
LGTM!

File: doc/source/whatsnew/v0.22.0.txt [hunk 5]
```diff
@@ -193,6 +193,7 @@ Other API Changes
 - Rearranged the order of keyword arguments in :func:`read_excel()` to align with :func:`read_csv()` (:issue:`16672`)
 - :func:`pandas.merge` now raises a ``ValueError`` when trying to merge on incompatible data types (:issue:`9780`)
 - :func:`wide_to_long` previously kept numeric-like suffixes as ``object`` dtype. Now they are cast to numeric if possible (:issue:`17627`)
+- comment arg is exposed as a named parameter in :func:`read_excel`
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: reverse this statement, ``in :func:`read_excel`, the ``comment`` argument is now exposed as a named parameter`` (and need double-backticks)

**Review by jreback — APPROVED** [review]
lgtm. just a small doc comments. ping on green.

**Comment by JanLauGe** [comment]
uhoh, seems I mocked something up.
Sorry @jreback, can you explain to me why is AppVeyor suddenly failing now?

File: doc/source/whatsnew/v0.22.0.txt [hunk 6]
```diff
@@ -198,6 +198,7 @@ Other API Changes
 - Rearranged the order of keyword arguments in :func:`read_excel()` to align with :func:`read_csv()` (:issue:`16672`)
 - :func:`pandas.merge` now raises a ``ValueError`` when trying to merge on incompatible data types (:issue:`9780`)
 - :func:`wide_to_long` previously kept numeric-like suffixes as ``object`` dtype. Now they are cast to numeric if possible (:issue:`17627`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: rebase on master. can you move to 0.23 (docs were renamed), prob easiest to just check this file from master and past in new one

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/excel.py [hunk 7]
```diff
@@ -164,6 +168,80 @@
...
+
+Examples
+--------
+
+An example DataFrame written to a local file
+
+>>> df_out = pd.DataFrame([('string1', 1),
+...                        ('string2', 2),
+...                        ('string3', 3)],
+...                       columns=('Name', 'Value'))
```
Comment (jorisvandenbossche) [inline pandas/io/excel.py]: please use a list for the column names (both are OK, but a list is IMO more idiomatic as tuples are also used for single labels of multi-indexed columns

File: pandas/io/excel.py [hunk 8]
```diff
@@ -137,7 +137,7 @@
 na_values : scalar, str, list-like, or dict, default None
     Additional strings to recognize as NA/NaN. If dict passed, specific
     per-column NA values. By default the following values are interpreted
-    as NaN: '""" + fill("', '".join(sorted(_NA_VALUES)), 70) + """'.
+    as NaN: '""" + fill("', '".join(sorted(_NA_VALUES)), 999) + """'.
```
Comment (jorisvandenbossche) [inline pandas/io/excel.py]: why this change?

File: pandas/io/excel.py [hunk 9]
```diff
@@ -164,6 +168,80 @@
...
+
+>>> pd.read_excel(open('tmp.xlsx','rb'))
+      Name  Value
+0  string1      1
+1  string2      2
+2  string3      3
+
+Index and header can be specified via the `index_col` and `header` arguments
+
+>>> pd.read_excel(open('tmp.xlsx','rb'), index_col=None, header=None)
```
Comment (jorisvandenbossche) [inline pandas/io/excel.py]: Can you use here (and in the following ones as well) just the "'tmp.xlsx'" (without the open). It is fine to show this ability (above), but I would use the simpler case for the rest of the examples

File: pandas/io/excel.py [hunk 10]
```diff
@@ -164,6 +168,80 @@
...
+0  string1    1.0
+1  string2    2.0
+2  string3    3.0
+
+True, False, and NA values, and thousands separators have defaults,
+but can be explicitly specified, too. Supply the values you would like
+as strings or lists of strings!
+
+>>> pd.read_excel(open('tmp.xlsx','rb'),
+...               true_values='2',
```
Comment (jorisvandenbossche) [inline pandas/io/excel.py]: This has no effect on the example?

File: pandas/io/excel.py (hunk 8 above)
Comment (jorisvandenbossche) [reply pandas/io/excel.py]: Ah, I see you mentioned this in the very first post of the PR. But, I don't think this is the fully correct solution, as this will make the line too long. 
Apparently for the `read_csv` docstring this is working properly, so maybe check how they did it there?

File: pandas/io/excel.py [hunk 11]
```diff
@@ -223,7 +301,8 @@ def read_excel(io,
                parse_dates=False,
                date_parser=None,
                thousands=None,
-               skipfooter=0,
+               comment=None,
+               skip_footer=0,
```
[the remark this answers is not in the document]
Comment (JanLauGe) [reply pandas/io/excel.py]: thanks

File: pandas/io/excel.py (hunk 8 above)
Comment (JanLauGe) [reply pandas/io/excel.py]: I think that the long line is not a problem, as line breaks in the input doc string do not translate to line breaks in the compiled output. I have tested compiling the docs locally and the output looked fine for me. However, since you referred to the read_csv doc string, I had a look at that. It uses `, subsequent_indent="    "`, so I will change it here to keep things consistent.

File: pandas/io/excel.py (hunk 10 above)
Comment (JanLauGe) [reply pandas/io/excel.py]: You're right! `true_values` is not working as I expected. Did I misunderstand the argument or is it broken?

File: pandas/io/excel.py [hunk 12]
```diff
@@ -132,12 +132,13 @@
 nrows : int, default None
     Number of rows to parse
 
-    .. versionadded:: 0.23.0
+    .. versionadded:: 0.22.0
```
Comment (jreback) [inline pandas/io/excel.py]: revert this (0.22 is a special release)

File: pandas/io/excel.py [hunk 13]
```diff
@@ -148,9 +149,13 @@
...
     any numeric columns will automatically be parsed, regardless of display
     format.
+comment : str, default None
+    Comments out remainder of line. Pass a character or characters to this
+    argument to indicate comments in the input file. Any data between the
+    comment string and the end of the current line is ignored.
 skip_footer : int, default 0
 
-    .. deprecated:: 0.23.0
+    .. deprecated:: 0.22.0
```
Comment (jreback) [inline pandas/io/excel.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
pushed a fix for the lint issue. ping on green.

**Comment by jreback** [comment]
thanks @JanLauGe
